core/jobs/io/firebase_io.py

```
@@ -0,0 +1,151 @@
+        settings_list = tuple(grouped['settings'])
+        auth_details_list = tuple(grouped['auth_details'])
+
+        try:
+            strictly_zipped = zip(settings_list, auth_details_list, strict=True)
+            [(settings, auth_details)] = strictly_zipped
+        except ValueError as e:
+            raise ValueError(
+                f'Oppia users need EXACTLY ONE of each model, but {user_id=!r} '
+                f'has {len(settings_list)} UserSettingsModels and '
+                f'{len(auth_details_list)} UserAuthDetailsModels'
+            ) from e
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)